lib/validate/validate_property.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+var validate = require('./validate');
+var ValidationError = require('../error/validation_error');
+var getType = require('../util/get_type');
+
+module.exports = function validateProperty(options, propertyType) {
+    var key = options.key;
+    var style = options.style;
+    var styleSpec = options.styleSpec;
+    var value = options.value;
+    var propertyKey = options.objectKey;
+    var layerSpec = styleSpec[propertyType + '_' + options.layerType];
```

Comment:
Gotcha. So far we've avoided using template literals and other ES6 features in mapbox-gl-style-spec. If we were to start, we'd need to think about the downstream implications. (This applies to the monorepo too. cc @lucaswoj)